src/core/Core.test.js

```diff
@@ -719,36 +719,61 @@ describe('src/Core', () => {
 
   describe('actions', () => {})
 
-  describe('isOnline', () => {
-    xit('should detect if the browser is online or not', () => {})
-
-    xit(
-      'should emit the correct event when the online status is specified',
-      () => {
-        const onlineEventMock = jest.fn()
-        const offlineEventMock = jest.fn()
-        const backOnlineEventMock = jest.fn()
-        const core = new Core()
-        core.on('is-offline', offlineEventMock)
-        core.on('is-online', onlineEventMock)
-        core.on('back-online', backOnlineEventMock)
-
-        core.isOnline(true)
-        expect(onlineEventMock.mock.calls.length).toEqual(1)
-        expect(offlineEventMock.mock.calls.length).toEqual(0)
-        expect(backOnlineEventMock.mock.calls.length).toEqual(0)
-
-        core.isOnline(false)
-        expect(onlineEventMock.mock.calls.length).toEqual(1)
-        expect(offlineEventMock.mock.calls.length).toEqual(1)
-        expect(backOnlineEventMock.mock.calls.length).toEqual(0)
-
-        core.isOnline(true)
-        expect(onlineEventMock.mock.calls.length).toEqual(2)
-        expect(offlineEventMock.mock.calls.length).toEqual(1)
-        expect(backOnlineEventMock.mock.calls.length).toEqual(1)
-      }
-    )
+  describe('updateOnlineStatus', () => {
+    const RealNavigatorOnline = global.window.navigator.onLine
+
+    beforeEach(() => {
+      Object.defineProperty(
+        global.window.navigator,
+        'onLine',
+        (function (_value) {
+          return {
+            get: function _get () {
+              return _value
+            },
+            set: function _set (v) {
+              _value = v
+            }
+          }
+        })(global.window.navigator.onLine)
```

Comment:
i think this would be a bit clearer without the IIFE, with the `_value` as a simple variable inside `beforeEach()`

Reply:
Agreed, though unfortunately I hit issues overriding the value of window.navigator.onLine (I think it's because it's a symbol and only has a getter). This was the solution I found on stackoverflow for mocking window.navigator properties

Reply:
Ah, gotcha. Would this work though?

```js
beforeEach(() => {
  let value = navigator.onLine
  Object.defineProperty(...)
})
```